### PR TITLE
Added Jenkins file

### PR DESCRIPTION
**Pull request by shaundmorris**
This PR adds Jenkins builds to the DDF-Admin-UI repo

@TonyMorrison 
@LinkMJB 
@AdamBurstynski 
@codice/build 
@codice/continuous-integration

---
This change is [Reviewable](https://reviewable.io/reviews/codice/ddf-admin-ui/6)

### CODE REVIEW

**Review by LinkMJB — COMMENTED** [review]
Should we add support for PR builds?

**Comment by shaundmorris** [comment]
> Should we add support for PR builds?

@LinkMJB yes I believe so. I was thinking at this point the branch builds and PR build would be the same. What is the best way to go about this?

**Comment by LinkMJB** [comment]
> > Should we add support for PR builds?
> 
> @LinkMJB yes I believe so. I was thinking at this point the branch builds and PR build would be the same. What is the best way to go about this?

@shaundmorris gotcha, I think what you have here should be sufficient then, I think it will accomplish the desired goal!

**Comment by LinkMJB** [comment]
```
~ 🍺
matthew.bates@LinkMJB $ validate_jenkinsfile https://raw.githubusercontent.com/codice/ddf-admin-ui/cb74f53d43f070028de392a08fd2e6f4af0b4a48/Jenkinsfile
...Downloading Jenkinsfile
...Validating Jenkinsfile
Jenkinsfile successfully validated.
```

**Review by LinkMJB — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        stage('Setup') {
+            steps {
+                dockerd {}
+                slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+                postCommentIfPR("Internal build has been started, your results will be available at build completion.", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+            }
+        }
+
+        // Checkout the repository
+        stage('Checkout repo') {
```
Comment (AdamBurstynski) [inline Jenkinsfile]: Not sure we need this stage for a new repository. 
If you remove the `disableDefaultCheckout()` from the options block the declarative pipeline will checkout the repository for you.

Comment (shaundmorris) [reply Jenkinsfile]: Good point. Do we need the retry logic? Do you know if we get that with the declarative pipeline?

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(BRANCH_NAME == "master" ? "H H(17-19) * * *" : "")
+    }
+    environment {
+        LARGE_MVN_OPTS = '-Xmx4G -Xms1G -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/sdk/**/*,**/*.js,**/node_modules/**/*,**/jaxb/**/*,**/wsdl/**/*,**/nces/sws/**/*,**/*.adoc,**/*.txt,**/*.xml'
```
Comment (shaundmorris) [inline Jenkinsfile]: I can remove this variable since it isn't used at all.

File: Jenkinsfile (hunk 1 above)
Comment (TonyMorrison) [reply Jenkinsfile]: I dont know if "retry" is automatic by default in the declarative pipeline. With a smaller repo do you suspect that we will timeout in checking out code?

Comment (shaundmorris) [reply Jenkinsfile]: I would hope we would not need it on the small repo. I'll try removing and add it back later if we need it.

**Review by TonyMorrison — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (shaundmorris) [reply Jenkinsfile]: I removed this stage and the skipDefaultCheckout() line

Comment (AdamBurstynski) [reply Jenkinsfile]: Looks good to me!
The logic to checkout the repo can be modified to ensure that only a single attempt is needed to clone.

**Review by AdamBurstynski — APPROVED** [review]

**Pull request merged**